fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/PersistentStorageTransaction.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to DuraSpace under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * DuraSpace licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.kernel.impl;
+
+
+/**
+ * Interface for persistent storage transaction implementation.
+ * 
+ * TODO: Implement javax.transaction.xa.XAResource
+ * javax.transaction:javax.transaction-api:1.3
+ *
+ * @author whikloj
+ * @since 2019-09-19
+ */
+public interface PersistentStorageTransaction {
+
+    /**
+     * Prepares the transaction to be committed by ensuring it can be locked.
+     */
+    public void prepare();
```

Comment:
I believe this method should throw a checked exception since the client would want to respond to possible conditions:  timeout waiting for locking mechanism, etag changed, etc.